bridgev2/networkinterface.go

```diff
@@ -688,6 +688,17 @@ type PowerLevelHandlingNetworkAPI interface {
 	HandleMatrixPowerLevels(ctx context.Context, msg *MatrixPowerLevelChange) (bool, error)
 }
 
+type MatrixJoinRulesChange struct {
+	MatrixRoomMeta[*event.JoinRulesEventContent]
+	OrigJoinRule event.JoinRule
+	NewJoinRule  event.JoinRule
+}
```

Comment:
This type is redundant, all the info is already in `JoinRulesEventContent`, so `type MatrixJoinRulesChange = MatrixRoomMeta[*event.JoinRulesEventContent]` is sufficient (similar to name/topic/avatar changes)